src/main/java/core/basesyntax/service/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.ReportGenerator;
+import core.basesyntax.service.ShopService;
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private final ShopService shopService;
+
+    public ReportGeneratorImpl(ShopService shopService) {
+        this.shopService = shopService;
+    }
+
+    @Override
+    public String getReport() {
+        StringBuilder report = new StringBuilder("fruit,quantity\n");
+        for (Map.Entry<String, Integer> entry : shopService.getStorage().entrySet()) {

```

Comment:
Directly accessing the storage map from the shopService violates the encapsulation principle. Consider adding a method in the ShopService that returns a copy of the storage data or the data necessary for the report generation, rather than exposing the internal storage structure.